js/script.js

```diff
@@ -17,13 +17,20 @@ const addButtonHandler = () => {
   } else showToastMessage("Please provide a valid title!");
 };
 
+const deleteTask = (taskId) => {
+  tasks = tasks.filter((task) => task.id !== taskId);
+  renderTasks();
+};
+
 const createTask = (taskTitle) => {
   const task = {
     id: new Date().getTime(),
     title: taskTitle,
     createdAt: formatDate(new Date()),
+    deleteButton: createElement("Delete", "button", () => deleteTask(task.id)),
```

Comment:
deleteButton is not part of the `task` data structure. It's a part of the `task` component.

The idea is:
```js
function createTaskElement(task){
  const $task = createElement('div')
  const $text = createElement('p')
  const $deleteButton = createElement('button')
  
  $text.textContent = task.text
  $deleteButton.addEventListener(someEventListener)
  
  $task.appendChild($text)
  $task.appendChild($deleteButton)

  return $task
}

// script.js
$taskList.innerHTML = ""

tasks.forEach(task => {
  const $task = createTaskElement(task)
  
  $taskList.appendChild($task)
})
```

Reply:
resolved, bhai.